test-suite/output/Fixpoint.out

```diff
@@ -35,3 +35,89 @@ File "./output/Fixpoint.v", line 75, characters 0-48:
 Warning: Not a fully mutually defined cofixpoint
 (a2 and a1 are not mutually dependent).
  [non-full-mutual,fixpoints,default]
+File "./output/Fixpoint.v", line 93, characters 2-15:
+The command has indeed failed with message:
+Recursive definition of foo and bar is ill-formed.
+As a mutual fixpoint decreasing on the 1st argument of foo and
+1st argument of bar:
+Not enough abstractions in the definition.
+As a mutual fixpoint decreasing on the 1st argument of foo and
+2nd argument of bar:
+Not enough abstractions in the definition.
+As a mutual fixpoint decreasing on the 2nd argument of foo and
+1st argument of bar:
+Not enough abstractions in the definition.
+As a mutual fixpoint decreasing on the 2nd argument of foo and
+2nd argument of bar:
+Not enough abstractions in the definition.
```

Comment:
this seems like it would get very long and unreadable as the number of possibilities increases.
It's already kinda bad IMO.

OTOH I don't know how to avoid listing the possibilities since the user didn't say `{struct}` (also I think there's currently no syntax for `{struct}` for `Lemma with`).

Reply:
With a bit more work, we could factorize the errors which are similar.

On the other side, #19301 adds syntax `{struct}` for `Lemma with` and for `Lemma`. So, an alternative could be to suggest using `struct` to get more information (???).

Reply:
We could also extend the Guarded syntax to be able to say eg `Guarded {struct 1, 3} {struct 3, 1}` to mean "test guardedness with 1st fixpoint recursive on 1st arg and 2nd on 3rd arg, and with 1st fixpoint recursive on 3rd arg and 2nd on 1st arg"
that way in the middle of a big proof there is no need to restart it to get a better error

Reply:
> this seems like it would get very long and unreadable as the number of possibilities increases.

> With a bit more work, we could factorize the errors which are similar.

I pushed a new version simplifying a bit the printing so that in the "Not enough abstractions", it is printed only once.

Reply:
> We could also extend the Guarded syntax to be able to say eg Guarded {struct 1, 3} {struct 3, 1}

That's a good idea.

To depart not too much from the existing syntax, we could also say, e.g.: `Guarded {struct 1} {struct 3}` and repeat it for any combination we are interested in??

Reply:
yes

Reply:
Do you want me to implement it? (And otherwise, I'd be happy if ever you have some time for it.)